server/aws-lsp-codewhisperer/src/language-server/chat/chatController.test.ts

```diff
@@ -249,5 +254,96 @@ describe('ChatController', () => {
                 })
             )
         })
+
+        describe('#extractEditorState', () => {
+            const typescriptDocument = TextDocument.create('file:///test.ts', 'typescript', 1, 'test')
+            let extractEditorStateStub: sinon.SinonStub
+            const editorStateObject = {}
+            const mockCursorState = {
+                range: {
+                    start: {
+                        line: 1,
+                        character: 1,
+                    },
+                    end: {
+                        line: 1,
+                        character: 1,
+                    },
+                },
+            }
+
+            beforeEach(() => {
+                extractEditorStateStub = sinon.stub(DocumentContextExtractor.prototype, 'extractEditorState')
+                testFeatures.workspace.getTextDocument.resolves(typescriptDocument)
```

Comment:
`TestFeatures` has an implementation of `getTextDocument` already you can leverage by using `openDocument` (https://github.com/aws/language-server-runtimes/blob/fe773e98f92d49890d6a72fd145b9192196ba51c/runtimes/testing/TestFeatures.ts#L96).

Saves you keeping track of another stub, and allows you to have multiple files open.

Reply:
that's great to know, I updated to use that